src/shared/shutdown.js

```diff
@@ -51,6 +58,15 @@ export const shutdownGateway: (logger: Logger) => Promise<void> = (logger) =>
                 );
             }
         });
+
+        // If the server hasn't shutdown in 15s then we force it to shutdown
+        // This could be because of connections being kept alive
+        setTimeout(() => {
+            logger.error(
+                "Server failed to close. Forcing it to shutdown.",
+                () => reject(),
+            );
+        }, 15000);
```

Comment:
Should this timeout be configurable so that we can easily tweak it in settings from the webapp implementation?